src/go/rpk/pkg/cli/connect/uninstall.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package connect
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/plugin"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func uninstallCommand(fs afero.Fs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "uninstall",
+		Short: "Uninstall the Redpanda Connect plugin",
+		Args:  cobra.NoArgs,
+		Run: func(cmd *cobra.Command, _ []string) {
+			connect, pluginExists := plugin.ListPlugins(fs, plugin.UserPaths()).Find("connect")
+			if !pluginExists {
+				out.Exit("The Redpanda Connect managed plugin is not installed!")
+			}
+			ops, anyFailed := removePluginAll(connect)
+			tw := out.NewTable("PATH", "MESSAGE")
+			defer func() {
+				tw.Flush()
+				if anyFailed {
+					os.Exit(1)
+				}
+			}()
+			for _, o := range ops {
+				tw.Print(o.path, o.message)
+			}
+		},
+	}
+	return cmd
+}
+
+type removeOperation struct {
+	path    string
+	message string
+}
+
+func removePluginAll(p *plugin.Plugin) (ops []removeOperation, anyFailed bool) {
```

Comment:
Do we do this same type of remove-all-and-shadow in byoc?

Reply:
Yes, moved this logic to the plugin method and re-used it in `byoc uninstall`